provd/persist/util.py

```diff
@@ -564,8 +564,8 @@ def func(document):
         cur_elem = document
         try:
             for cur_key in split_key:
-                cur_elem = cur_elem[cur_key]
-        except (KeyError, TypeError) as e:
+                cur_elem = cur_elem.get(cur_key, '') or ''
```

Comment:
Sorry, again. I don't know if this is possible, but `or ''` will also happen for "fasley" values. Such as `[]`, `{}`, `0` and `False`. I'm not sure what data this can possible contain, but it might be an issue. Instead of the `or` at the end you could do `return '' if cur_elem is None else str(cur_elem)` ?

Reply:
I updated the code. 
Now if the value is not a str/int/float -> the str(value) is put at the end of the list.